src/main/java/org/sopt/service/MemberServiceImpl.java

```diff
@@ -0,0 +1,68 @@
+package org.sopt.service;
+
+import org.sopt.domain.Member;
+import org.sopt.dto.member.MemberSignupRequest;
+import org.sopt.exception.DuplicateEmailException;
+import org.sopt.exception.MemberNotFoundException;
+import org.sopt.exception.UnderageMemberException;
+import org.sopt.repository.MemoryMemberRepository;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.Period;
+import java.util.List;
+import java.util.Optional;
+
+public class MemberServiceImpl implements MemberService {
+
+    private final MemoryMemberRepository memberRepository = new MemoryMemberRepository();
+    private static long sequence = 1L;
+
+    @Override
+    public Long join(MemberSignupRequest request) {
+
+        if(existsByEmail(request.getEmail())) {
+            throw new DuplicateEmailException();
+        }
+
+        if(calculateAge(request.getBirthDate()) < 20) {
+            throw new UnderageMemberException();
+        }
+
+        Member member = new Member(
+                sequence++,
+                request.getName(),
+                request.getEmail(),
+                request.getGender(),
+                request.getBirthDate()
+        );
+        memberRepository.save(member);
+        return member.getId();
+    }
+
+    @Override
+    public Optional<Member> findOne(Long memberId) {
+        return memberRepository.findById(memberId);
+    }
+
+    @Override
+    public List<Member> findAllMembers() {
+        return memberRepository.findAll();
+    }
+
+    @Override
+    public void delete(Long memberId) {
+        if(!memberRepository.deleteById(memberId)){
+            throw new MemberNotFoundException();
+        }
+    }
+
+    private Boolean existsByEmail(String email) {
+        return memberRepository.existsByEmail(email);
+    }
```

Comment:
private 함수로 검증하는 부분 빼신 것 너무 좋습니다! 다만 지금 메서드가 리포지토리의 existsByEmail을 호출하는 것과 같은 역할을 한다고 생각해서 의미 중복인 것 같아요.
아래처럼 바꾸면 검증 로직 자체를 해당 메서드가 담당할 수 있을 것 같습니다!
```suggestion
private void validateIsUniqueEmail(String email) {
    if(memberRepository.existsByEmail(email)) {
        throw new DuplicateEmailException();
    }
}